hdfs-scheduler/src/main/java/org/apache/mesos/hdfs/scheduler/HdfsMesosConstraints.java

```diff
@@ -0,0 +1,107 @@
+package org.apache.mesos.hdfs.scheduler;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.apache.mesos.Protos.Attribute;
+import org.apache.mesos.Protos.Offer;
+import org.apache.mesos.Protos.Value.Range;
+import org.apache.mesos.hdfs.config.HdfsFrameworkConfig;
+
+/**
+ * HDFS Mesos offer constraints checker class implementation.
+ */
+public class HdfsMesosConstraints {
+
+  private final HdfsFrameworkConfig config;
+
+  public HdfsMesosConstraints(HdfsFrameworkConfig config) {
+    this.config = config;
+  }
+
+  public boolean constraintsAllow(Offer offer) {
+    List<Attribute> attributes = offer.getAttributesList();
+    
+    Map<String, String> constraints = config.getMesosSlaveConstraints();
+    Set<Map.Entry<String, String>> constraintSet = constraints.entrySet();
+    
+    for (Map.Entry<String, String> constraintEntry : constraintSet) {
+      boolean found = false;
+      String constraintName = constraintEntry.getKey();
+      String constraintValue = constraintEntry.getValue();
+      
+      for (Attribute attribute : attributes) {
+        if (attribute.getName().equals(constraintName)) {
+          switch (attribute.getType()) {
+          case RANGES:
+            if (attribute.hasRanges()) {
+              try {
+                Long range = Long.parseLong(constraintValue);
+                for (Range r : attribute.getRanges().getRangeList()) {
+                  if ((!r.hasBegin() || range >= r.getBegin())
+                      && (!r.hasEnd() || range <= r.getEnd())) {
+                    found = true;
+                    break;
+                  }
```

Comment:
if neither begin nor end is set, it means that it is open range and match should only be on basis of attribute name. So found must be true